src/providers/mux/provider.ts

```diff
@@ -27,25 +29,27 @@ function sleep(ms: number) {
 }
 
 async function pollForAssetReady(filePath: string, asset: Asset) {
-  if (!asset.externalIds?.assetId) {
+  if (!asset.providerSpecific?.[provider]?.assetId) {
```

Comment:
Probably overthinking this, but this is sprinkled throughout this file, would it clean this up to move the getter logic up top instead of the `provider` var?

```typescript
// Maybe in a utils file?
function getProviderValue(provider, asset, key) {
  return asset.providerSpecific?.[provider]?.[key];
}

// or we could take the overthinking further and specify a curried getter?
const providerValueFetcher = (provider) => (asset, key) => asset.providerSpecific?.[provider]?.[key];

// then use it in this file like:
const getAssetValue = providerValueFetcher('mux');
getAssetValue('assetId');
```

Reply:
I tried it out but caching the obj lookup seemed simplest.
we only have 2 lines like this now

`const providerSpecifics: MuxSpecifics | undefined = asset.providerSpecific?.mux;`

also added types